vaccine_feed_ingest/stages/enrichment.py

```diff
@@ -52,6 +54,45 @@ def enrich_locations(input_dir: pathlib.Path, output_dir: pathlib.Path) -> bool:
     return True
 
 
-def _process_location(loc: location.NormalizedLocation) -> location.NormalizedLocation:
-    """Run throuch all of the methods to enrich the location"""
-    return loc
+def _process_location(
+    normalized_location: location.NormalizedLocation,
+) -> Optional[location.NormalizedLocation]:
+    """Run through all of the methods to enrich the location"""
+    enriched_location = normalized_location.copy()
+
+    _add_provider_from_name(enriched_location)
+
+    return enriched_location
+
+
+def _generate_link_map(loc: location.NormalizedLocation) -> Dict[str, str]:
+    """Return a map of authority to id value"""
+    return {
+        str(link.authority): link.id
+        for link in loc.links or []
+        if link.authority and link.id
+    }
+
+
+def _add_provider_from_name(loc: location.NormalizedLocation) -> None:
+    """Add provider link from name if missing"""
+    if not loc.name:
+        return
+
+    provider_tuple = normalize.provider_id_from_name(loc.name)
+
+    if not provider_tuple:
+        return
+
+    provider_authority, provider_id = provider_tuple
+
+    existing_links = _generate_link_map(loc)
+
+    if str(provider_authority) not in existing_links:
+        loc.links = [
+            *(loc.links or []),
+            location.Link(authority=provider_authority, id=provider_id),
+        ]
```

Comment:
Should this _also_ add an `authority` if the `provider_authority` does exist but is set to a different ID?

I could see arguments for this either way.

Reply:
I would prefer to play it safe since this is automatically added it.

If the normalizer added a different id then the normalizer may have had a good reason for it, and i want to allow the normalizer to have override privileges here.